### PR TITLE
Add READMEs for spark tasks and timestamp task

### LINKED ISSUE #16: Add README for Spark job
_From @sabbyanandan on February 16, 2016 15:45_

As a developer, I'd like to add Spark docs, so it can be used as a reference for anyone wanting to run Spark jobs.

**Acceptance:**
- Ref. guide includes `spark` section
- Example included

_Copied from original issue: spring-cloud/spring-cloud-task-modules#27_

### PR DESCRIPTION
**Pull request by trisberg**
Fixes #16

### CODE REVIEW

File: spring-cloud-starter-task-spark-client/README.adoc [hunk 1]
```diff
@@ -1,25 +1,22 @@
 //tag::ref-doc[]
 = Spark Client Task
 
-TODO
+This task is intended to launch a Spark application from a task. The task submits the Spark application for local execution. This task is appropriate for a local deployment where any local file references can be resolved. It is not meant for any type of cluster deployments.
```
Comment (cppwfs) [inline spring-cloud-starter-task-spark-client/README.adoc:4]: First sentence should read: `This task is intended to launch a Spark application.` Will fix on merge

File: spring-cloud-starter-task-spark-yarn/README.adoc [hunk 2]
```diff
@@ -0,0 +1,35 @@
+//tag::ref-doc[]
+= Spark YARN Task
+
+This task is intended to launch a Spark application from a task. The task submits the Spark application to a YARN clusterfor execution. This task is appropriate for a deployment that has access to a Hadoop YARN cluster. The Spark application jar and the Spark Assembly jar should be referenced from an HDFS location.
```
Comment (cppwfs) [inline spring-cloud-starter-task-spark-yarn/README.adoc:4]: clusterfor change to `cluster for`.  Will fix on merge

File: spring-cloud-starter-task-spark-cluster/README.adoc [hunk 3]
```diff
@@ -0,0 +1,23 @@
+//tag::ref-doc[]
+= Spark Cluster Task
+
+This task is intended to launch a Spark application from a task. The task submits the Spark application for execution in a Spark cluster. This task is appropriate for a deployments where any file references can be resolved to a shared location.
```
Comment (cppwfs) [inline spring-cloud-starter-task-spark-cluster/README.adoc:4]: First sentence should read: This task is intended to launch a Spark application. Will fix on merge

File: spring-cloud-starter-task-spark-cluster/README.adoc [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+== Options
+
+// see syntax (soon to be automatically generated) in spring-cloud-stream starters
+The **$$spark-client$$** $$task$$ has the following options:
```
Comment (cppwfs) [inline spring-cloud-starter-task-spark-cluster/README.adoc:9]: copy paste error `spark-client` should be `spark-cluster`

File: spring-cloud-starter-task-spark-cluster/README.adoc [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+$$spark.app-jar$$:: $$The path to a bundled jar that includes your application and its dependencies, excluding any Spark dependencies.$$ *($$String$$, default: `$$<none>$$`)*
+$$spark.app-name$$:: $$The name to use for the Spark application submission.$$ *($$String$$, default: `$$<none>$$`)*
+$$spark.executor-memory$$:: $$The memory setting to be used for each executor.$$ *($$String$$, default: `$$1024M$$`)*
+$$spark.master$$:: $$The master setting to be used (spark://host:port).$$ *($$String$$, default: `$$spark://localhost:7077$$`)*
+$$spark.resource-archives$$:: $$A comma separated list of archive files to be included with the app submission.$$ *($$String$$, default: `$$<none>$$`)*
+$$spark.resource-files $$:: $$A comma separated list of files to be included with the application submission.$$ *($$String$$, default: `$$<none>$$`)*
+$$spark.rest-url$$:: $$The URL for the Spark REST API to be used (spark://host:port).$$ *($$String$$, default: `$$spark://localhost:6066$$`)*
+//end::configuration-properties[]
+
+//end::ref-doc[]
```
Comment (cppwfs) [inline spring-cloud-starter-task-spark-cluster/README.adoc:23]: Need to add `$$spark.app-status-poll-interval$$::  $$The interval (ms) to use for polling for the App status.$$ *($$long$$, default:`$$1000L$$`)*

Will fix on merge.

Comment (cppwfs) [inline spring-cloud-starter-task-spark-cluster/README.adoc:23]: Need to add $$spark.rest-urll$$:: $$The URL for the Spark REST API to be used (spark://host:port).$$ _($$String$$, default:$$spark://localhost:6066$$`)_

will fix on merge

File: spring-cloud-starter-task-spark-yarn/README.adoc (hunk 2 above)
Comment (cppwfs) [inline spring-cloud-starter-task-spark-yarn/README.adoc:4]: First sentence should read: This task is intended to launch a Spark application. Will fix on merge

File: spring-cloud-starter-task-spark-yarn/README.adoc [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+== Build
+
+```
+$ mvn clean package
+```
+
+== Run
+
+```
+$ java -jar target/spark-client-task-${version}-exec.jar
```
Comment (cppwfs) [inline spring-cloud-starter-task-spark-yarn/README.adoc:34]: copy/paste hiccup should read `spark-yarn-task`

Will fix on merge

Comment (cppwfs) [inline spring-cloud-starter-task-spark-yarn/README.adoc:34]: we removed the build and execution commands from the other spark apps.  Do we want this here or do we remove it.?

Comment (trisberg) [inline spring-cloud-starter-task-spark-yarn/README.adoc:34]: remove any build/execution stuff

File: spring-cloud-starter-task-spark-cluster/README.adoc (hunk 5 above)
Comment (trisberg) [inline spring-cloud-starter-task-spark-cluster/README.adoc:23]: hmm, that's the last one already -- $$spark.rest-urll$$

Comment (trisberg) [inline spring-cloud-starter-task-spark-cluster/README.adoc:23]: we do need $$spark.app-status-poll-interval$$, thanks, missed that one

File: spring-cloud-starter-task-spark-client/README.adoc (hunk 1 above)
Comment (trisberg) [inline spring-cloud-starter-task-spark-client/README.adoc:4]: thx

File: spring-cloud-starter-task-spark-yarn/README.adoc (hunk 2 above)
Comment (trisberg) [inline spring-cloud-starter-task-spark-yarn/README.adoc:4]: yes, thx

File: spring-cloud-starter-task-spark-cluster/README.adoc (hunk 3 above)
Comment (trisberg) [inline spring-cloud-starter-task-spark-cluster/README.adoc:4]: yep

File: spring-cloud-starter-task-spark-cluster/README.adoc (hunk 4 above)
Comment (trisberg) [inline spring-cloud-starter-task-spark-cluster/README.adoc:9]: good catch

**Comment by cppwfs** [comment]
LGTM.  Fixed requested changes above on merge.  Rebased, squashed, merged.

**Pull request closed without merging**